library/core/src/option.rs

```diff
@@ -90,6 +90,321 @@
 //! from `Some::<T>(_)` to `T` (but transmuting `None::<T>` to `T`
 //! is undefined behaviour).
 //!
+//! # Method overview
+//!
+//! In addition to working with pattern matching, [`Option`] provides a wide
+//! variety of different methods.
+//!
+//! ## Querying the variant
+//!
+//! The [`is_some`] and [`is_none`] methods return [`true`] if the [`Option`]
+//! is [`Some`] or [`None`], respectively.
+//!
+//! [`is_some`]: Option::is_some
+//! [`is_none`]: Option::is_none
+//!
+//! ## Adapters for working with references
+//!
+//! * [`as_ref`] converts from `&Option<T>` to `Option<&T>`
+//! * [`as_mut`] converts from `&mut Option<T>` to `Option<&mut T>`
+//! * [`as_deref`] converts from `&Option<T>` to `Option<&T::Target>`
+//! * [`as_deref_mut`] converts from `&mut Option<T>` to `Option<&mut T::Target>`
+//! * [`as_pin_ref`] converts from [`&Pin`]`<Option<T>>` to `Option<`[`Pin`]`<&T>>`
```

Comment:
This is a minor nit, but would it make sense to write this as:

```suggestion
//! * [`as_pin_ref`] converts from `&`[`Pin`]`<Option<T>>` to `Option<`[`Pin`]`<&T>>`
```

(And analogously for `&mut Pin`.)

That would avoid making the `&` a link (which seems misleading), and would allow just defining one link to `Pin` rather than three.

Reply:
Thanks! I double-checked against the method definitions, and there were actual errors in my text, so I fixed those as well. (It's `Pin<&Option<T>>`, not `&Pin<Option<T>>`, etc.)